src/interactive.rs

```diff
@@ -558,6 +572,9 @@ async fn _interactive_main(options: &Options, state: &mut repl::State)
         state.ensure_connection()
             .race(ctrlc.wait_result())
             .await?;
+        if let Some(conn) = state.connection.as_mut() {
+            update_idle_transaction_timeout(conn).await?;
+        }
```

Comment:
We need to adjust it after every reconnect. It should be somewhere in `state.ensure_connection` I think.